packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -184,6 +213,84 @@ function getZeroIndexStateText(subjectNoun?: string) {
   return "No options yet";
 }
 
+function useComboboxAccessibility(
+  open: boolean,
+  selectionCallback: (selection: ComboboxOption) => void,
+  containerRef: React.RefObject<HTMLDivElement>,
+  filteredOptions: ComboboxOption[],
+  optionsListRef: React.RefObject<HTMLUListElement>,
+): void {
+  const hasOptionsVisible = open && filteredOptions.length > 0;
+  const [focusedIndex, setFocusedIndex] = useState<number | null>(null);
+
+  useEffect(() => setFocusedIndex(null), [open, filteredOptions.length]);
+
+  useEffect(() => {
+    if (open) {
+      containerRef.current?.addEventListener("keydown", handleContentKeydown);
+    }
+
+    return () => {
+      containerRef.current?.removeEventListener(
+        "keydown",
+        handleContentKeydown,
+      );
+    };
+  }, [open, optionsListRef, focusedIndex, filteredOptions]);
```

Comment:
this is a bit aggressive, but what happens without these dependencies is that the handler added to the event gets stale. it's almost a snapshot of the state when we attached it, but we need up to date versions of these vars when doing the logic in `handleContentKeydown`

I feel like there's gotta be a better way to do this but I'm blanking at the moment